modules/library/main/src/main/java/org/geotools/feature/visitor/UniqueVisitor.java

```diff
@@ -166,23 +222,33 @@ public void reset() {
 
     @Override
     public CalcResult getResult() {
-        if (set.isEmpty()) {
-            return CalcResult.NULL_RESULT;
-        }
-        return new UniqueResult(set, this.preserveOrder);
+        if (set.isEmpty()) return CalcResult.NULL_RESULT;
+        else return new UniqueResult(set, this.preserveOrder, attrNames);
+    }
+
+    public List<String> getAttrNames() {
+        return attrNames;
     }
 
     public static class UniqueResult extends AbstractCalcResult {
+
+        private List<String> attributeNames;
```

Comment:
I see the attributeNames being carried around in both the UniqueVisitor and the UniqueResult as new fields.
The bits in UniqueResults do not seem to be used. The ones in UniqueVisitor seem to be used in tests, but do not seem to be mandatory, e.g., the code building the UniqueVisitor already knows the name, and in JDBCDataStore, but again only as an expedient to know how many attrbutes there are (getExpressions() should cover the same need then).

So wondering, could they be removed?

Reply:
fixed